.github/workflows/package.yml

```diff
@@ -0,0 +1,51 @@
+name: npm-package-build
+
+# Only trigger on:
+# - master branch
+# - PR or Pull Request event types
+# - Exclide Docusaurus files: this file, docs/** and website/**
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - '.github/workflows/docusaurus.yml'
```

Comment:
Here there should probably be the following paths:
```
- src
- test
- package.json
- tsconfig.json
- yarn.lock
```

Same in the block below.

Reply:
@maoo I did consider this, but I think it's not ideal. It's easy to think of changes to other files than the ones you mentioned above, that could impact the build (or adding new files at the root level that could, e.g. an eslint file).

README.md goes into the npm package, for example. Similarly changes to the license file, or editor config. And if the build files themselves change.

So I went with the "exclude docusaurus files/folders and include everything else" approach.